js/init.js

```diff
@@ -55,14 +58,24 @@ function highlightFeature( e ) {
 }
 
 function resetHighlight( e ) {
-	if( selected == undefined || e.target.feature.properties.id != selected.feature.properties.id ) neighborhoods.resetStyle( e.target );
+	if( $.isEmptyObject( selected ) || e.target.feature.properties.id != selected.feature.properties.id ) neighborhoods.resetStyle( e.target );
 	$( "#probe" ).hide();
 }
 
 function featureClick( e ) {
-	if( selected ) neighborhoods.resetStyle( selected );
+	var executing = $( this );
+	if ( executing.data( 'executing' ) ) return;
+	executing.data( 'executing', true );
+	
+	if( $.isEmptyObject( selected ) || e.target.feature.properties.id != selected.feature.properties.id ) {
+		neighborhoods.resetStyle( selected );
+		$( '#name-input' ).val('');
+	}
 	selected = e.target;
-	show_details( e.target.feature.properties );
+	selected.setStyle({
+		color : '#ed2a24'
+	});
+	show_details( e.target.feature.properties, executing );
```

Comment:
Identifier 'show_details' is not in camel case.<br>'show_details' is not defined.